openwisp_monitoring/device/admin.py

```diff
@@ -116,6 +116,27 @@ def __init__(self, *args, **kwargs):
             }
         super().__init__(*args, **kwargs)
 
+    def save(self, commit=True):
+        instance = super().save(commit=False)
+        # If all the fields are empty,
+        # then just delete the object
+        # to prevent the default value
+        # from being used as a fallback
+        if all(
+            fields is None
```

Comment:
```suggestion
            not fields
```

Let's write is this way to also account for empty strings.